lib/std/os/bits/linux.zig

```diff
@@ -1515,3 +1515,77 @@ pub const rusage = extern struct {
     nivcsw: isize,
     __reserved: [16]isize = [1]isize{0} ** 16,
 };
+
+pub const cc_t = u8;
+pub const speed_t = u32;
+pub const tcflag_t = u32;
+
+pub const NCCS = 32;
+
+pub const IGNBRK = 1;
+pub const BRKINT = 2;
+pub const IGNPAR = 4;
+pub const PARMRK = 8;
+pub const INPCK = 16;
+pub const ISTRIP = 32;
+pub const INLCR = 64;
+pub const IGNCR = 128;
+pub const ICRNL = 256;
+pub const IUCLC = 512;
+pub const IXON = 1024;
+pub const IXANY = 2048;
+pub const IXOFF = 4096;
+pub const IMAXBEL = 8192;
+pub const IUTF8 = 16384;
+
+pub const OPOST = 1;
+pub const OLCUC = 2;
+pub const ONLCR = 4;
+pub const OCRNL = 8;
+pub const ONOCR = 16;
+pub const ONLRET = 32;
+pub const OFILL = 64;
+pub const OFDEL = 128;
+pub const VTDLY = 16384;
+pub const VT0 = 0;
+pub const VT1 = 16384;
+
+pub const CSIZE = 48;
+pub const CS5 = 0;
+pub const CS6 = 16;
+pub const CS7 = 32;
+pub const CS8 = 48;
+pub const CSTOPB = 64;
+pub const CREAD = 128;
+pub const PARENB = 256;
+pub const PARODD = 512;
+pub const HUPCL = 1024;
+pub const CLOCAL = 2048;
+
+pub const ISIG = 1;
+pub const ICANON = 2;
+pub const ECHO = 8;
+pub const ECHOE = 16;
+pub const ECHOK = 32;
+pub const ECHONL = 64;
+pub const NOFLSH = 128;
+pub const TOSTOP = 256;
+pub const IEXTEN = 32768;
+
+pub const TCSA = extern enum(usize) {
```

Comment:
This should really be `c_int` as it's part of the libc definitions.

Reply:
Well in my defense I was advised to get rid of the awkward casting in the `tcsetattr` function (this was originally `i32`).

Reply:
You did fine, @nofmal. We're just using this opportunity to consider how to further improve the maintainability / organization of this code.

The "bits" that are libc-specific (which it seems like applies here) should go in `lib/std/c/linux.zig`. 

Is this really libc-specific though? It looks like this enum is relevant to applications that do not link against libc as well.

Reply:
This works with and without libc enabled. I mean the goal of this PR was to not depend on libc in the first place.

Reply:
Seems like status quo is correct then.

Reply:
This is used in the definition of `tcsetattr` in `c.zig`, the libc prototype expects an `int` there.

Reply:
OK, I agree that it should be set up such that the type is c_int when interfacing with the libc functions.

This is where I think @daurnimator is a bit over-zealous with non-exhaustive enums, and we're better off just doing it with `pub const foo = bar;` declarations like it's done in C.

Reply:
The enum can use `c_uint` then :)

Reply:
`c_uint` is inappropriate for this when not linking libc. It's planned for C types and C integration to be an "optional extension" to the zig language specification. A zig compiler without the C integration extension would report a compile error for using the `c_uint` integer type.